### PR TITLE
[PEFT] Set eval mode when loading PEFT adapter

### LINKED ISSUE #34469: PeftAdapterMixin.load_adapter takes the model out of eval()
https://github.com/huggingface/transformers/blob/c2820c94916e34baf4486accae74760972183a2f/src/transformers/integrations/peft.py#L71

When you load a trained adapter using `load_adapter` it takes the model out of eval mode: I would expect this behavior from the `add_adapter` method, but not the loading method. 

```
model_dir = "/path/to/seq2seq/model"
tokenizer = AutoTokenizer.from_pretrained(model_dir)
model:PeftAdapterMixin = AutoModelForSeq2SeqLM.from_pretrained(model_dir).to(device)
model.eval()
for module in model.modules():
    assert module.training == False # This will not throw an exception
model.load_adapter(adapter_path)
for module in model.modules():
    assert module.training == False # This will throw an exception
```

### PR DESCRIPTION
**Pull request by BenjaminBossan**
# What does this PR do?

When calling `model.load_adapter` to load a PEFT adapter, by default the adapter should be set to eval mode. This is now correctly done. Users can still pass `is_trainable=True` to load the adapter in training mode.

Fixes #34469

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
@ArthurZucker could you please review or suggest a reviewer?

**Comment by BenjaminBossan** [comment]
Another ping @ArthurZucker for review or reviewer suggestion.

**Review by Rocketknight1 — COMMENTED** [review]
Doing an initial review until we can find a core maintainer - the code and tests look good, but maybe we should default `is_trainable` to the model's current state - in other words, the arg `is_trainable` could default to `None`, and if the user doesn't manually set a value, then we set it equal to `model.training`. This means we'll load the adapter in trainable mode if `model.train()` is set but not if `model.eval()` is set.

Maybe there's a reason that's a bad idea, though, I'm not sure!

**Comment by BenjaminBossan** [comment]
Thanks for the review @Rocketknight1.

Regarding your suggestion about inferring `is_trainable` from the model. This would only make a difference for the case where the user explicitly puts the model into training mode, right? However, this would be highly unusual, as with PEFT, you typically want the base weights to be frozen and only train the PEFT weights. Therefore, I don't think this would help in a real world use case. And for the rare occasion that the user wants all parameters to be trainable for whatever reason, they could just call `model.train()` after calling `load_adapter`.

This doesn't mean that your suggestion can't be implemented, it would just add a bit of complexity where it's probably not needed. LMK what you think.

**Review by ArthurZucker — APPROVED** [review]
LGTM but this seems to be breaking no? If so let's put a 🔴 on the PR title !
